src/appengine/handlers/cron/fuzzer_and_job_weights.py

```diff
@@ -326,6 +339,35 @@ def store_current_weights_in_bigquery():
   client.insert(rows)
 
 
+def update_job_weight(job_name, multiplier):
+  """Update a job weight."""
+  tool_name = environment.get_memory_tool_name(job_name)
+  weight = SANITIZER_WEIGHTS.get(tool_name, DEFAULT_SANITIZER_WEIGHT)
+  weight = weight * multiplier
+
+  query = data_types.FuzzerJob.query(data_types.FuzzerJob.job == job_name)
+  changed_weights = []
+  for fuzzer_job in query:
+    if fuzzer_job.automated_weight != weight:
+      fuzzer_job.automated_weight = weight
+      changed_weights.append(fuzzer_job)
+
+  if changed_weights:
+    ndb.put_multi(changed_weights)
+
+
+def update_job_weights():
+  """Update job weights."""
+  for job in data_types.Job.query():
+    multiplier = DEFAULT_MULTIPLIER
+    if environment.is_engine_fuzzer_job(job.name):
+      targets_count = ndb.Key(data_types.FuzzTargetsCount, job.name).get()
+      if targets_count:
+        multiplier = targets_count.count
```

Comment:
I am a little bit worried about this. So, a project like ffmpeg with >400 targets should have 400 weight ? isnt that too high, should we have a cap so that cycles are not taken up by one project. (maybe a test for this ?)

Reply:
Why not? FFmpeg is a huge project with dozens of dependencies. If the goal of this code alone is to allocate the same number of cycles for every individual fuzz target, I'm against any capping or e.g. logarithmic weight calculation.

Same logic applies for the internal stuff. If a directory has a lot of fuzz targets, why should they get fewer cycles each?

Reply:
And we still have manually assigned weights at our discretion. Any super critical stuff can get a higher weight in there and each fuzz target will get more CPU cycles than the default.

Reply:
+1 to Max's comments. I think cases where we don't want to treat targets as equal to one another are the uncommon ones, and we can just use manual weights if we ever feel like a particular project is using too many CPUs.